### ISSUE TITLE
FIXED pid and msgid for idempotence producer

### ISSUE THREAD

**Issue by calvin2021y** [edenhill/librdkafka#3648]
With enable.idempotence=true,  I want to pass pid and msgid into rdkafka.  Is there a way to do it ?

For fixed pid, because there could be multi producer share same pid,  there are not work at same time but cloud have small time overlap switch from A to B.

**Comment by PratRanj07** [comment]
No, librdkafka does not support setting custom Producer IDs - the PID is automatically obtained from the broker via InitProducerIdRequest and is managed entirely internally. For producer identity across restarts/switchovers, use transactional.id configuration instead, which provides proper PID management and automatic fencing of stale producers.
I am closing this now as this is a bit old. If you have any more questions related to it. Please consider reopening this issue or creating a new one.
Thanks!
